bitnami/solr/templates/statefulset.yaml

```diff
@@ -16,6 +16,7 @@ metadata:
 spec:
   podManagementPolicy: {{ .Values.podManagementPolicy }}
   replicas: {{ .Values.replicaCount }}
+  revisionHistoryLimit: {{ .Values.revisionHistoryLimit }}
```

Comment:
```suggestion
{{ if .Values.revisionHistoryLimit }}
  revisionHistoryLimit: {{ .Values.revisionHistoryLimit }}
 {{- end }}
```

Reply:
The default value for revisionHistoryLimit is 10, so my change proposal is not a breaking change, it keeps the defaults. Putting this inside an if block or not setting the value, does not "unset" the value at the end in kubernetes.

BUT: if the user sets the value to 0 as an integer (which is natural to me), then the if condition will be false.
The revisionHistoryLimit won't be set by helm, so kubernetes will use the default value for revisionHistoryLimit which is... 10. A user setting revisionHistoryLimit to 0 (integer) in values.yaml will at the end then get 10. :face_with_spiral_eyes: 
This would be a bug to me.
I would recommend against the if block and the empty value in the values.yaml

Reply:
You are right about the default value, but if "you set this field to 0, will result in cleaning up all the history of your Deployment thus that Deployment will not be able to roll back"
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#clean-up-policy.

It would be nice if you put something about the value 0, in values.yaml to clarify things.